scripts/verify_adv_IM.py

```diff
@@ -32,79 +66,166 @@ def parse_args():
         help="list of component that ran",
     )
 
+    parser.add_argument(
+        "--station_file",
+        default=None,
+        type=str,
+        help="if set, script will only check for folder-names that match the station file",
+    )
+
+    # saves output
+    parser.add_argument(
+        "--simple_check",
+        default=None,
+        action="store_true",
+        help="a quick comparison of station list, will not generate status.csv",
+    )
+
     args = parser.parse_args()
 
     return args
 
 
-def check_log(list_folders, model, csv_stations, components):
+def check_log(list_folders, model, components, df_model, break_on_fail=False):
     """
     check of "Failed" msg in logs
     no "Failed" in all Analysis*.txt == crashed
     "Failed" in all Analysis*.txt == failed to converge (normal)
     """
-    failed_stations = []
     for station_dir in list_folders:
         # check if folder is a station run folder
         station_model_dir = os.path.join(station_dir, model)
         station_name = os.path.basename(station_dir)
-        if not os.path.isdir(station_model_dir) or (station_name in csv_stations):
-            # does not match naming for a station run, possibly created by something else, skipping
-            continue
         for comp in components:
             component_outdir = os.path.join(station_model_dir, comp)
-            if not check_status(component_outdir, check_fail=True):
-                failed_stations.append(station_name)
-                # logs shows not all analysis failed to converge, but station failed to create csv
-                print(f"{station_dir} failed to run on {model}")
-                # break the loop as soon as one component fail the verification
-                break
-    return failed_stations
-
+            time_list = read_timelog(component_outdir)
+            if check_status(component_outdir):
+                # successed
+                station_component_status = run_status.finished.value
+            # check for 'Failed' keyword
```

Comment:
Be consistent with comments before or after conditional check (or even in-line)

Reply:
sorry i dont really get it. do you mean i should be consistent on if the comment for certain check is before or after the line?

Reply:
Yep. Especially because this one is duplicated below the line as well as above the line.